code/controllers/subsystem/mapping.dm

```diff
@@ -184,17 +184,22 @@ SUBSYSTEM_DEF(mapping)
 				if(index)
 					lists_to_reserve.Cut(1, index)
 				return
-			var/turf/T = packet[packetlen]
-			T.empty(RESERVED_TURF_TYPE, RESERVED_TURF_TYPE, null, TRUE)
-			LAZYINITLIST(unused_turfs["[T.z]"])
-			unused_turfs["[T.z]"] |= T
-			var/area/old_area = T.loc
-			LISTASSERTLEN(old_area.turfs_to_uncontain_by_zlevel, T.z, list())
-			old_area.turfs_to_uncontain_by_zlevel[T.z] += T
-			T.turf_flags = UNUSED_RESERVATION_TURF
-			world_contents += T
-			LISTASSERTLEN(world_turf_contents_by_z, T.z, list())
-			world_turf_contents_by_z[T.z] += T
+			var/turf/reserving_turf = packet[packetlen]
+			reserving_turf.empty(RESERVED_TURF_TYPE, RESERVED_TURF_TYPE, null, TRUE)
+			LAZYINITLIST(unused_turfs["[reserving_turf.z]"])
+			unused_turfs["[reserving_turf.z]"] |= reserving_turf
+			var/area/old_area = reserving_turf.loc
+			LISTASSERTLEN(old_area.turfs_to_uncontain_by_zlevel, reserving_turf.z, list())
+			old_area.turfs_to_uncontain_by_zlevel[reserving_turf.z] += reserving_turf
+			reserving_turf.turf_flags = UNUSED_RESERVATION_TURF
+
+			// reservation turfs are not allowed to interact with atmos at all
+			reserving_turf.blocks_air = TRUE
+			reserving_turf.immediate_calculate_adjacent_turfs()
```

Comment:
you should not need to do this, since empty() queues a recalc and this does not cancel that out. if you want to be safe use the macro that queues one up

Reply:
similar to below